### ISSUE TITLE
blink does not blink on a Pico W

### ISSUE THREAD

**Issue by repentsinner** [raspberrypi/pico-examples#421]
The Pico W 'hello world' (aka blink) out-of-the-box experience is terrible as of [sdk-1.5.1](https://github.com/raspberrypi/pico-examples/tree/sdk-1.5.1).

Please add at least an additional `#warning if using a pico w, blink has moved to pico-examples/pico_w/wifi/blink` in `pico-examples\blink\blink.c`. Preferred would be to update `blink.c` to include both the vanilla and W code paths. I suspect beginners to the Pico (W)(H) platform using the C SDK will be able to figure out what is going on, and it could be a good place to include some comments about other potential gotcha differences between the two boards.

While truly the LED is now connected to the wireless chip and therefore pedantically sample code for it should be in the `pico-examples/pico_w` directory, I highly doubt anyone looking for `blink` cares about that distinction when looking for a first thing to run to figure out if their board is working _at all_.

The official [overview documentation](https://www.raspberrypi.com/documentation/microcontrollers/raspberry-pi-pico.html#raspberry-pi-pico-w-and-pico-wh) has a few W-specific notes; none of them describe the moved LED. It is only section 3.8 of [the Pico W datasheet](https://datasheets.raspberrypi.com/picow/pico-w-datasheet.pdf) that gets around to rather dryly describing this move that then *might* point someone in the correct direction in the examples directory:

> There are three GPIO pins from the CYW43439 that are used for other board functions and can easily be accessed via the SDK:
> [...]
> WL_GPIO0
> OP connected to **user LED**

#350 has drifted off topic to somewhat address this, but it seems like a new issue is more appropriate.

**Comment by lurch** [comment]
@aallan I've just had another look at Chapter 3 of https://datasheets.raspberrypi.com/pico/getting-started-with-pico.pdf and it has this boxout on page 9
![Screenshot from 2023-09-17 15-01-15](https://github.com/raspberrypi/pico-examples/assets/476186/1629adf0-ed37-4001-a4d3-47d4beb4f731)
but it's not very clear that the user actually needs to compile and build a **different** `blink` example program if they're using a Pico W board rather than a Pico board.

**Comment by peterharperuk** [comment]
> Preferred would be to update blink.c to include both the vanilla and W code paths. 

Yes, I forget how we got into this situation. We should probably do something as it seems to be causing confusion.

**Comment by XeYaRiQu** [comment]
@repentsinner Thank you for opening this issue as I am a beginner and have just experienced what you described. I followed with the "Getting Started with Raspberry Pi Pico" document and assumed that the Pico and Pico W were the same except for the addition of the WiFi/BT chip. Panicked for a moment thinking I had a defective Pico W.

**Comment by aallan** [comment]
This will be fixed in the next documentation release, where the two blink examples are explicitly addressed in the _Getting Started_ book.

**Comment by repentsinner** [comment]
Thanks so much @peterharperuk!

**Comment by yoyojacky** [comment]
![image](https://github.com/user-attachments/assets/ec9e012d-463b-4d81-b087-f70d0cf79549)
![image](https://github.com/user-attachments/assets/11ea7398-cdc4-41b1-ba6a-fc268d2e8655)
what should i do ? replace it with what kind of header file?

**Comment by peterharperuk** [comment]
To include cyw43_arch.h you need to link to pico_cyw43_arch_none

**Comment by lurch** [comment]
@yoyojacky `cyw43_arch.h` isn't in `pico-sdk/lib/cyw43-driver/src/` because it's actually in `pico-sdk/src/rp2_common/pico_cyw43_arch/include/pico/` :wink: 

If you're having problems compiling the `blink` example though, I suggest you create a new issue describing your problem in detail, rather than commenting on an already-closed issue.
